text_extensions_for_pandas/array/test_tensor.py

```diff
@@ -419,6 +419,57 @@ def test_numpy_properties(self):
         self.assertEqual(arr.numpy_shape, data.shape)
         self.assertEqual(arr.numpy_dtype, data.dtype)
 
+    def test_bool_tensor_selection(self):
+        data = TensorArray([[1, 2], [3, 4], [5, 6]])
+        sel = TensorArray([True, False, True])
+        expected = np.array([[1, 2], [5, 6]])
+
+        # Test TensorArray.__getitem__ with TensorArray
+        result = data[sel]
```

Comment:
You should probably test what happens when `sel` is a 2D tensor. Is that case supposed to return an error, or is it supposed to perform numpy indexing?

Reply:
Yeah, good idea